### PR TITLE
Bottom Nav fixes for external control

### PR DESCRIPTION
**Pull request by GiantRobots**
…allow the escape button to close the nav.

### How was this tested 👻 

- [ ] Manual testing
- [ ] E2E tests added
- [ ] Unit tests added

### CODE REVIEW

**Review by Alex-Tideman — APPROVED** [review]

File: src/lib/components/bottom-nav.svelte [hunk 1]
```diff
@@ -27,6 +28,33 @@
...
 
+  const escHandler = new AbortController();
+
+  onMount(() => {
+    document.addEventListener(
+      'keydown',
+      (e) => {
+        if (
+          e.key === 'Escape' &&
+          [viewSettings, viewNamespaces, viewSettings].some((isOpen) => isOpen)
```
Comment (rossedfort) [inline src/lib/components/bottom-nav.svelte:39]: `viewSettings` is specified here twice

File: src/lib/components/bottom-nav.svelte [hunk 2]
```diff
@@ -27,6 +28,33 @@
   let viewNamespaces = writable(false);
   let viewSettings = false;
 
+  const escHandler = new AbortController();
```
Comment (rossedfort) [inline src/lib/components/bottom-nav.svelte:31]: What is the purpose of this? I see it added in the event listener, but where is it referenced, in a `fetch` somewhere?

Comment (GiantRobots) [reply src/lib/components/bottom-nav.svelte:31]: You can use abortControllers to remove event listeners now. https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#signal

File: src/lib/components/bottom-nav.svelte (hunk 1 above)
Comment (GiantRobots) [reply src/lib/components/bottom-nav.svelte:39]: Wisdom

**Review by rossedfort — APPROVED** [review]

**Pull request merged**